strangerstudios/paid-memberships-pro
# Improving the pmpro_generateUsername function.

### All Submissions:






### Changes proposed in this Pull Request:


The pmpro_generateUsername function would fail if no first or last name was passed into the function. Additionally, it was using a direct database query where a core WP function could have been used.

### How to test the changes in this Pull Request:
This PR is primarily designed to support a customized checkout process where the username and password fields are removed. 

1. Add the following recipe to your PMPro Customizations plugin: https://gist.github.com/kimcoleman/75022e78bc2b245b6265127519f67248
2. Make sure Register Helper Add On is installed and activated.
3. Attempt checkout for a free level, providing only your email address.
4. The function will generate a username based on the email address provided.

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
* BUG FIX/ENHANCEMENT: Refactored the pmpro_generateUsername function to allow for email only signup or completely random username generation.
* ENHANCEMENT: Now using the wp_generate_password function to generate a user's password at checkout.
